python/ray/_private/runtime_env/pip.py

```diff
@@ -167,20 +168,32 @@ async def _check_ray(python: str, cwd: str, logger: logging.Logger):
         """
 
         async def _get_ray_version_and_path() -> Tuple[str, str]:
-            check_ray_cmd = [
-                python,
-                "-c",
-                "import ray; print(ray.__version__, ray.__path__[0])",
-            ]
-            if _WIN32:
-                env = os.environ.copy()
-            else:
-                env = {}
-            output = await check_output_cmd(
-                check_ray_cmd, logger=logger, cwd=cwd, env=env
-            )
-            # print after import ray may have [0m endings, so we strip them by *_
-            ray_version, ray_path, *_ = [s.strip() for s in output.split()]
+            with tempfile.TemporaryDirectory() as tmp_dir:
```

Comment:
Do we need to change to use `tempfile.NamedTemporaryFile(suffix="_ray_version.txt") `?